src/mapreduce.jl

```diff
@@ -160,25 +160,76 @@ end
 ## reducedim ##
 ###############
 
-@inline reducedim(op, a::StaticArray, ::Val{D}) where {D} = mapreducedim(identity, op, a, Val{D})
-@inline reducedim(op, a::StaticArray, ::Val{D}, v0) where {D} = mapreducedim(identity, op, a, Val{D}, v0)
+@inline reducedim(op, a::StaticArray, ::Type{Val{D}}) where {D} = mapreducedim(identity, op, a, Val{D})
```

Comment:
OK, so what I would like is if *StaticArrays* used `Val{D}` everywhere in v0.6 and `Val(D)` everywhere in v0.7. (I suppose it's OK if the v0.6 version also supports `Val{D}()`, partly to avoid this being a breaking change)

If you were to implement that here, that would be awesome (as seems pertinent to the change already done on this line) - however, let me know if you'd prefer to not do that at the moment, and I'll get to it later.